File: 3426-MinimumNumberOfChairsInAWaitingRoom/3426-MinimumNumberOfChairsInAWaitingRoom.java
// Last updated: 12/28/2025, 4:52:03 PM
class Solution {
    public int minimumChairs(String s) {
        int c=0,ans=0;
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i)=='E')
            {
                c++;
                ans=Math.max(ans,c);
            }
            else
            {
                c--;
            }
        }
        return ans;
    }
}

File: 3426-MinimumNumberOfChairsInAWaitingRoom/3426-MinimumNumberOfChairsInAWaitingRoom.py
# Last updated: 12/28/2025, 4:52:06 PM
class Solution:
    def minimumChairs(self, s: str) -> int:
        c=0
        ans=0
        for i in s:
            if i=='E':
                c+=1
                ans=max(ans,c)
            else:
                c-=1
        return ans